src/tribler/core/components/gigachannel_manager/tests/test_gigachannel_manager.py

```diff
@@ -222,6 +224,65 @@ def mock_process_channel_dir(c, _):
     assert not gigachannel_manager.channels_processing_queue
 
 
+async def test_check_channel_updates_for_different_states(gigachannel_manager, metadata_store):
```

Comment:
This function can be safely converted to a simple sync function, which is better because it allows you to avoid logic involving async loop creation:
```suggestion
def test_check_channel_updates_for_different_states(gigachannel_manager, metadata_store):
```

Reply:
good point